bundle/Resources/config/services.yml

```diff
@@ -295,6 +296,17 @@ services:
         class: %ezpublish_legacy.fieldType.ezimage.variation_service.class%
         arguments: [@ezpublish_legacy.kernel]
 
+    # core Index listener override
+    ezpublish.request_index_listener:
+        class: %ezpublish_legacy.index_request_listener.class%
+        arguments:
+            - @ezpublish.config.resolver
+            - @router
+            - %ezpublish.siteaccess.default%
+            - @?logger
+        tags:
+            - { name: kernel.event_subscriber }
+
```

Comment:
Why do you create a new service? All you have to do is to override `ezpublish.request_index_listener.class` parameter